flake.nix

```diff
@@ -0,0 +1,24 @@
+{
+  description = "A tool for generating images of code and terminal output";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-23.11";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = {
+    self,
+    nixpkgs,
+    flake-utils,
+  }:
+    flake-utils.lib.eachDefaultSystem (system: let
+      pkgs = import nixpkgs {inherit system;};
+    in {
+      packages.default = import ./default.nix {inherit pkgs;};
+    })
```

Comment:
you could remove flake-utils here by using the snippit below. It would also eliminate the slower `import`.

```nix
}: let
    forAllSystems = nixpkgs.lib.genAttrs ["x86_64-linux" "x86_64-darwin" "i686-linux" "aarch64-linux" "aarch64-darwin"];
    
    pkgsForEach = nixpkgs.legacyPackages;
  in {
    packages = forAllSystems (system: {
      default = pkgsForEach.${system}.callPackage ./default.nix {};
    });
```

Reply:
After a bit of testing a better overlay would be.

```nix
 overlays.default = final: prev: {
   charm-freeze = final.callPackage ./default.nix {};
 };
```